src/account/mod.rs

```diff
@@ -88,8 +97,163 @@ pub struct Account {
     /// Transaction payloads for received outputs with inputs when not pruned before syncing, can be used to determine
     /// the sender address/es
     #[serde(rename = "incomingTransactions")]
-    incoming_transactions: HashMap<TransactionId, (TransactionPayload, Vec<OutputWithMetadataResponse>)>,
+    #[serde(deserialize_with = "deserialize_or_convert")]
+    incoming_transactions: HashMap<TransactionId, Transaction>,
     /// Foundries for native tokens in outputs
     #[serde(rename = "nativeTokenFoundries", default)]
     native_token_foundries: HashMap<FoundryId, FoundryOutput>,
 }
+
+// Custom deserialization to stay backwards compatible
+fn deserialize_or_convert<'de, D>(deserializer: D) -> Result<HashMap<TransactionId, Transaction>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let value = serde_json::Value::deserialize(deserializer)?;
+
+    type NewType = HashMap<TransactionId, Transaction>;
+    type OldType = HashMap<TransactionId, (TransactionPayload, Vec<OutputWithMetadataResponse>)>;
+
+    Ok(match serde_json::from_value::<NewType>(value.clone()) {
```

Comment:
Just checking but this isn't going to use a default value if it fails, right ? Otherwise it would always be `Ok`.

Reply:
If both types fail it will return an error 3 lines below

Reply:
Yeah I got that, I was just wondering if the first one can actually fail, if it doesn't default to an empty hashmap instead of failing since I think I was a default attribute somewhere.

Reply:
Just tried to deserialize with a "essence" in the payload renamed to "notworking" and it failed